pkg/apis/testharness/v1beta1/test_types.go

```diff
@@ -125,6 +125,9 @@ type TestStep struct {
 
 	// Kubeconfig to use when applying and asserting for this step.
 	Kubeconfig string `json:"kubeconfig,omitempty"`
+
+	// Loads kubeconfig only when the previous steps have finished.
+	LazyLoadKubeconfig bool `json:"lazyLoadKubeconfig,omitempty"`
```

Comment:
Please read the comment on bools in [this conventions doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) and turn this field into something like `kubeConfigLoading` with allowed values of `Eager` and `Lazy`.
Also please update the `./docs/` in this repo.